connectors/src/resources/microsoft_resource.ts

```diff
@@ -205,6 +208,8 @@ export class MicrosoftRootResource extends BaseResource<MicrosoftRootModel> {
   }
 }
 
+export type MicrosoftNodeType = WithCreationAttributes<MicrosoftNodeModel>;
```

Comment:
Seems weird we should not expose those types outside of the resource functions.

Reply:
This type was the premise of the "intermediary" type that you mention below :)